boa/src/builtins/object/gcobject.rs

```diff
@@ -0,0 +1,79 @@
+//! This module implements the `GcObject` structure.
+//!
+//! The `GcObject` is a garbage collected Object.
+
+use super::Object;
+use gc::{Finalize, Gc, GcCell, GcCellRef, GcCellRefMut, Trace};
+use std::fmt::{self, Display};
+
+/// Garbage collected `Object`.
+#[derive(Debug, Trace, Finalize, Clone)]
+pub struct GcObject(Gc<GcCell<Object>>);
+
+impl GcObject {
+    #[inline]
+    pub(crate) fn new(object: Object) -> Self {
+        Self(Gc::new(GcCell::new(object)))
+    }
+
+    #[inline]
+    pub fn borrow(&self) -> GcCellRef<'_, Object> {
+        match self.try_borrow() {
+            Ok(object) => object,
+            Err(e) => panic!("{}", e),
+        }
```

Comment:
It's not exactly, because we use `panic!("{}", e)`, this displays the structure with the `Display` trait (the implementation displays the error "object is already mutually borrowed")

The `std::cell::RefCell` `try_borrow` and `try_borrow_mut` just calls `.expect()` as well as having `Display` implementation for the error types `BorrowError` and `BorrowMutError`, I think I'll do the same.

`RefCell`s `borrow` method: https://doc.rust-lang.org/src/core/cell.rs.html#797-799